lib/use_cases/update_rent_charges.rb

```diff
@@ -0,0 +1,78 @@
+class UpdateRentCharges
+  def initialize(fixed_data_gateway:, rent_charges_gateway:)
+    @fixed_data_gateway = fixed_data_gateway
+    @rent_charges_gateway = rent_charges_gateway
+  end
+
+  def execute(fixed_data)
+    errors = validations(fixed_data)
+
+    return { successful: false, errors: errors } unless errors.empty?
+  
+    @fixed_data_gateway.save(fixed_data)
+    @rent_charges_gateway.save(rent_charges(fixed_data))
+
+    { successful: true }
+  end
+
+  private
+
+  def rent_charges(fixed_data)
+    rent_charges = @rent_charges_gateway.all((fixed_data[:year].to_i) - 1)
+    rent_charges.map do |rent_charge|
+      { 
+        uprn: rent_charge.uprn,
+        address: rent_charge.address,
+        address_2: rent_charge.address_2,
+        comments: rent_charge.comments,
+        rr_count: rent_charge.rr_count,
+        property_type: rent_charge.property_type,
+        base_data_bed_size: rent_charge.base_data_bed_size,
+        bedroom_weight: rent_charge.bedroom_weight,
+        mra_archetype: rent_charge.mra_archetype,
+        jan_1999_asset_values: rent_charge.jan_1999_asset_values,
+        year: rent_charge.year + 1,
```

Comment:
adding 1 because here we are building the rent charges for the next year, so we pull out the rent charges for the previous year and then perform the necessary calculations to update (including incrementing the year)